app/models/sitewide_setting.rb

```diff
@@ -17,11 +17,13 @@ def name
   def republish_downstream_if_reshuffle
     return unless key == "minister_reshuffle_mode"
 
-    payload = PublishingApi::MinistersIndexPresenter.new
-
-    Services.publishing_api.put_content(payload.content_id, payload.content)
-    Services.publishing_api.publish(payload.content_id, nil, locale: "en")
-
-    PresentPageToPublishingApiWorker.perform_async("PublishingApi::HowGovernmentWorksPresenter")
+    update_live = true
+    if on
+      PresentPageToPublishingApiWorker.perform_async("PublishingApi::HowGovernmentWorksEnableReshufflePresenter", update_live)
+      PresentPageToPublishingApiWorker.perform_async("PublishingApi::MinistersIndexEnableReshufflePresenter", update_live)
```

Comment:
I think we'll probably want to do something like this before merge:

```suggestion
      # send 'page is currently being updated' message to Draft and promote to Live
            PresentPageToPublishingApiWorker.perform("PublishingApi::HowGovernmentWorksEnableReshufflePresenter", update_live)
      PresentPageToPublishingApiWorker.perform("PublishingApi::MinistersIndexEnableReshufflePresenter", update_live)
      # Now send normal ministers index payload to draft so that we can use it as a preview
      PresentPageToPublishingApiWorker.perform("PublishingApi::MinistersIndexPresenter", false)
```

Reply:
That seems reasonable, if a bit annoying

Reply:
Just to double-check - you mean the "having to reorder a minister in order to see the preview" workaround is reasonable, rather than the code suggestion above?

Reply:
Sorry, I meant the code suggestion is reasonable. Just a bit annoying that we have to make an extra request and make them all synchronously, which is something our architecture is trying to help us not to have to do